lib/node_modules/@stdlib/math/base/ops/cneg/lib/main.js

```diff
@@ -20,45 +20,69 @@
 
 // MODULES //
 
-var negate = require( './cnegate.js' );
+var real = require( '@stdlib/complex/real' );
+var imag = require( '@stdlib/complex/imag' );
+var Complex128 = require( '@stdlib/complex/float64' );
 
 
 // MAIN //
 
 /**
-* Negates a complex number.
+* Negates a double-precision complex floating-point number.
 *
-* @param {(Array|TypedArray|Object)} [out] - output array
-* @param {number} re - real component
-* @param {number} im - imaginary component
-* @returns {(Array|TypedArray|Object)} negated components
+* @param {Complex128} z - complex number
+* @returns {Complex128} result
 *
 * @example
-* var v = cneg( -4.2, 5.5 );
-* // returns [ 4.2, -5.5 ]
+* var Complex128 = require( '@stdlib/complex/float64' );
+* var real = require( '@stdlib/complex/real' );
+* var imag = require( '@stdlib/complex/imag' );
+* var cneg = require( '@stdlib/math/base/ops/cneg' );
 *
-* @example
-* var out = new Array( 2 );
+* var z1 = new Complex128( -4.2, 5.5 );
+* // returns <Complex128>
 *
-* var v = cneg( out, -4.2, 5.5 );
-* // returns [ 4.2, -5.5 ]
+* var out = cneg( z1 );
+* // returns <Complex128>
 *
-* var bool = ( v === out );
-* // returns true
+* var re = real( out );
+* // returns 4.2
 *
-* @example
-* var v = cneg( 0.0, 0.0 );
-* // returns [ -0.0, -0.0 ]
+* var im = imag( out );
+* // returns -5.5
 *
-* @example
-* var v = cneg( NaN, NaN );
-* // returns [ NaN, NaN ]
+* var z2 = new Complex128( 0.0, 0.0 );
+* // returns <Complex128>
+*
+* var out = cneg( z2 );
+* // returns <Complex128>
+*
+* var re = real( out );
+* // returns -0.0
+*
+* var im = imag( out );
+* // returns -0.0
+*
+* var z3 = new Complex128( NaN, NaN );
+* // returns <Complex128>
+*
+* var out = cneg( z3 );
+* // returns <Complex128>
+*
+* var re = real( out );
+* // returns NaN
+*
+* var im = imag( out );
+* // returns NaN
 */
-function cneg( out, re, im ) {
-	if ( arguments.length === 2 ) {
-		return negate( [ 0.0, 0.0 ], out, re );
-	}
-	return negate( out, re, im );
+function cneg( z ) {
+	var re;
+	var im;
+
+	re = real( z );
+	im = imag( z );
+
+	return new Complex128( -re, -im );
```

Comment:
```suggestion
	return new Complex128( -real( z ), -imag( z ) );
```

Let's simplify a bit by not using extra var declarations.